waiter/integration/waiter/token_request_test.clj

```diff
@@ -1453,6 +1467,20 @@
             (assert-waiter-response response)
             (is (str/includes? body custom-maintenance-message))))
 
+        (testing "request to /waiter-ping should report error with custom maintenance message"
+          (let [{:keys [body] :as response}
+                (make-request waiter-url "/waiter-ping" :headers request-headers)]
+            (assert-response-status response http-200-ok)
+            (assert-waiter-response response)
+            (let [ping-body-json (try-parse-json body)
+                  ping-response-body-json (try-parse-json (get-in ping-body-json ["ping-response" "body"]))]
+              (is (= "waiter.Maintenance" (str (get-in ping-response-body-json ["waiter-error" "details" "error-class"]))))
+              (is (str/includes? (str (get-in ping-response-body-json ["waiter-error" "message"])) custom-maintenance-message))
+              (is (= "waiter-response" (get-in ping-body-json ["ping-response" "result"])))
+              (is (false? (get-in ping-body-json ["service-state" "exists?"])))
+              (is (false? (get-in ping-body-json ["service-state" "healthy?"])))
+              (is (= "Inactive" (get-in ping-body-json ["service-state" "status"]))))))
```

Comment:
We should also assert that the `ping-body-json` `:status` is a 503.

Reply:
Done.